### ISSUE TITLE
loader_unexpand_inst_layer_names writes through const pointer

### ISSUE THREAD

**Issue by Ralith** [KhronosGroup/Vulkan-LoaderAndValidationLayers#164]
During calls to `vkCreateInstance` and `vkCreateDevice`, `loader_unexpand_inst_layer_names` writes to the array passed in through the `const char* const* ppEnabledLayerNames` members of the respective CreateInfo structs. Although the written value seems to be exactly what was already there, when read-only memory is passed in ppEnabledLayerNames as the type permits, undefined behavior results. In my environment this consistently crashes my program in the loader.

The write occurs on loader.c:1749 in the assignment to *pp_str in this loop:
```C
    for (uint32_t i = 0; i < layer_count; i++) {
        char **pp_str = (char **)&pCreateInfo->ppEnabledLayerNames[i];
        *pp_str = layer_names[i];
    }
```

The following program consistently reproduces the crash:
```C
#include <vulkan/vulkan.h>

int main(int argc, char** argv) {
  const char* const layers[] = {
    "VK_LAYER_LUNARG_param_checker"
  };

  VkInstance instance;
  VkInstanceCreateInfo info;
  info.sType = VK_STRUCTURE_TYPE_INSTANCE_CREATE_INFO;
  info.pNext = NULL;
  info.flags = 0;
  info.pApplicationInfo = NULL;
  info.enabledLayerCount = 1;
  info.ppEnabledLayerNames = layers;
  info.enabledExtensionCount = 0;
  info.ppEnabledExtensionNames = NULL;
  vkCreateInstance(&info, NULL, &instance);
}
```

**Comment by chrisforbes** [comment]
@jon-lunarg @jeremy-lunarg this (and related badness) is addressed by #274

**Comment by jon-lunarg** [comment]
Chris fixed this issue, changes have been pushed to master.
